src/components/admin/Settings/StorageMappings/Dialog/useRepublishPrefix.ts

```diff
@@ -0,0 +1,113 @@
+    const { jobStatusPoller } = useJobStatusPoller();
+
+    const setPubId = useStorageMappingStore((state) => state.setPubId);
+    const setLogToken = useStorageMappingStore((state) => state.setLogToken);
+    const setSaving = useStorageMappingStore((state) => state.setSaving);
+    const setServerError = useStorageMappingStore(
+        (state) => state.setServerError
+    );
+
+    return useCallback(
+        async (prefix: string) => {
+            const republicationResponse = await republishPrefix(prefix);
+
+            if (republicationResponse.error) {
+                setSaving(false);
+                setServerError(republicationResponse.error);
+
+                logRocketEvent(CustomEvents.REPUBLISH_PREFIX_FAILED, {
+                    response: republicationResponse,
+                });
+
+                return;
+            }
+
+            if (!hasLength(republicationResponse.data)) {
+                setSaving(false);
+                setServerError(
+                    intl.formatMessage({
+                        id: 'storageMappings.dialog.generate.error.unableToFetchLogs',
+                    })
+                );
+
+                logRocketEvent(CustomEvents.REPUBLISH_PREFIX_FAILED, {
+                    context: 'A publication ID was not found in the response.',
+                });
+
+                return;
+            }
+
+            const evaluatedPubId = republicationResponse.data;
+            setPubId(evaluatedPubId);
+
+            const publicationResponse = await getPublicationById(
+                evaluatedPubId
+            );
+
+            if (publicationResponse.error || !publicationResponse.data) {
+                setSaving(false);
+                setServerError(
+                    intl.formatMessage({
+                        id: 'storageMappings.dialog.generate.error.unableToFetchLogs',
+                    })
+                );
+
+                logRocketEvent(CustomEvents.REPUBLISH_PREFIX_FAILED, {
+                    context: 'An error was encountered fetching the log token.',
+                });
+
+                return;
+            }
+
+            setLogToken(publicationResponse.data[0].logs_token);
+
+            jobStatusPoller(
+                supabaseClient
+                    .from(TABLES.PUBLICATIONS)
+                    .select(JOB_STATUS_COLUMNS)
+                    .eq('id', evaluatedPubId),
+                async () => {
+                    setSaving(false);
+                    setServerError(null);
+                },
+                async (payload: any) => {
+                    setSaving(false);
+                    setServerError(
+                        intl.formatMessage({
+                            id: 'storageMappings.dialog.generate.error.republicationFailed',
+                        })
+                    );
+
+                    logRocketEvent(CustomEvents.REPUBLISH_PREFIX_FAILED, {
+                        response: payload,
```

Comment:
You cannot pass an "open ended" object to eventing. LR will complain and against their terms. This should be like the other ones. If you want the more open ended logging you can then write a log rocket log right after this.